### PR TITLE
Add contributing docs and scripts/docs

### PR DESCRIPTION
**Pull request by yeraydiazdiaz**
Closes https://github.com/encode/httpcore/issues/95

Images are shared with HTTPX once https://github.com/encode/httpx/pull/1011 is merged.

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+welcome to [open an issue](https://github.com/encode/httpcore/issues/new).
+
+Please attach as much detail as possible and, in case of a
+bug report, provide information like:
+
+- OS platform or Docker image
+- Python version
+- Installed dependencies and versions (`python -m pip freeze`)
+- Code snippet to reproduce the issue
+- Error traceback and output
```
Comment (florimondmanca) [inline docs/contributing.md:33]: Okay with this, though in my experience people tend to actually not look at the contributing guide and forget to include these basic repro info bits. :-) But we can look at setting up issue templates as a follow up, though for now I think the volume is manageable…

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
+# Contributing
+
+Thanks for considering contributing to HTTP Core!
```
[the remark this answers is not in the document]
Comment (yeraydiazdiaz) [reply docs/contributing.md:3]: I went for consistency with what's in the [index](https://github.com/encode/httpcore/blob/master/docs/index.md), I'm happy to change it to `HTTPCore` though. @tomchristie ?

File: docs/contributing.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+We welcome contributors to:
+
+- Try [HTTPX](https://www.python-httpx.org), as it is HTTP Core's main entry point,
```
[the remark this answers is not in the document]
Comment (yeraydiazdiaz) [reply docs/contributing.md:7]: Those are slightly different, in my mind. "Entry point" underlines that you people should use HTTPX to interact with HTTP Core. Whereas "dependent" is a bit more broad?

File: docs/contributing.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+```shell
+$ cd httpcore
+$ scripts/install
+```
+
+## Unasync
+
+HTTP Core provides synchronous and asynchronous interfaces. As you can imagine,
+keeping two almost identical versions of code in sync can be quite time consuming.
+To work around this problem httpcore uses a technique called _unasync_, where
```
[the remark this answers is not in the document]
Comment (yeraydiazdiaz) [reply docs/contributing.md:70]: Ah that one slipped through, I'll change it to `HTTP Core` for now until we land on the form.

File: docs/contributing.md (hunk 1 above)
Comment (yeraydiazdiaz) [reply docs/contributing.md:33]: Sure, HTTPX also has this so I copied over for the most part.

File: docs/contributing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (florimondmanca) [reply docs/contributing.md:3]: Ah, okay, let's keep it like that for the purpose of this PR then :-)

**Comment by yeraydiazdiaz** [comment]
Thanks for the review @florimondmanca 👍

**Pull request merged**